deno_dist/adapter/deno/conninfo.ts

```diff
@@ -0,0 +1,18 @@
+import type { GetConnInfo } from '../../helper/conninfo/index.ts'
+
+/**
+ * Get conninfo with Deno
+ * @param c Context
+ * @returns ConnInfo
+ */
+export const getConnInfo: GetConnInfo = (c) => {
+  const { remoteAddr } = c.env
```

Comment:
This is funny.
But if you set up your own `env`, this will no longer work.
```ts
Deno.serve((req) => {
  const env = { url: "https://hono.dev"};
  return app.fetch(req, env);
});
```

Reply:
Yes, it's fault. Bun also so.
Bun's WebSocket helper doesn't work as well.

Should I validate `c.env`?

Reply:
`c.env` is environment variables that users should use freely, and I don't think it's a good idea to have an API that becomes unusable if you set env.
Although a common API to get an IP address is a great idea, I think we need to consider another way.

I don't have any good ideas at this point :'(

Reply:
Hmm..., surely.

But I think that c.env has created to get Cloudflare Workers's bindings, so this usage may be correct.

Also, you can make custom envs by the way:
```ts
Deno.serve((req, server) => {
  const env = { ...server, url: "https://hono.dev"};
  return app.fetch(req, env);
});
```
@yusukebe What do you think about that?

Reply:
@nakasyou 

You may use `env()` in `hono/adapter`!

https://hono.dev/helpers/adapter#env

Reply:
@yusukebe
This helper needs 2nd argument of fetch, it doesn't need runtime environment values.

Reply:
@nakasyou 

Sorry! I understood.

As you said, the second arg of `app.fetch` is designed for Cloudflare Workers's Bindings first. But now, as @ryuapp said, users can use it freely. The node-server append the `HttpBindings` to it:

https://github.com/honojs/node-server/blob/main/src/listener.ts#L186

Reply:
Hmm.

If we don't change the API of the core, this method @nakasyou said is a better way now.

```ts
Deno.serve((req, server) => {
  const env = { ...server, url: 'https://hono.dev' }
  return app.fetch(req, env)
})
```

Reply:
@yusukebe, thank you.

@ryuapp @yusukebe, Do you think that this problem can resolve if we add details of second argument of fetch to docs when pr merged?

Reply:
@nakasyou I think yes.

But, this Bun case seems not to work:

```ts
Bun.serve({
  fetch: (req: Request, server) => {
    const env = { ...server, foo: 'bar' }
    return app.fetch(req, env)
  },
})

// bun run src/index.ts
```

Could you try it?

Reply:
Hi @yusukebe.
Oh, I got error too because `server` is class instance.

What do you think about this code?:
```ts
Bun.serve({
  fetch: (req: Request, server) => {
    const env = { sever, foo: 'bar' }
    return app.fetch(req, env)
  },
})
```
In this code, if `c.env.requestIP` is undefined, helper tries to access `c.env.server.requestIP`.

I'll implement it if it's better.

Reply:
It's ok if we write that on docs.
I've reconsidered that the complexity of setting up env outweighs the benefits of this feature.
Thank you.

Reply:
@nakasyou 

> What do you think about this code?

Seems to be good! The naming `server` would be better because the Bun is naming the second argument of `fetch` as `server`:

![CleanShot 2024-05-06 at 22 19 50@2x](https://github.com/honojs/hono/assets/10682/2ae0275a-bdcf-4421-8dc9-a1b6dc665d2f)

https://bun.sh/docs/api/http#bun-serve

Reply:
@ryuapp @yusukebe thank you for reviewing!
I'll implement it!

Reply:
@nakasyou 

Gooood!